docs/tutorials/v8-migration-guide.mdx

```diff
@@ -0,0 +1,349 @@
+
+> `frameloop` will not be respected while in a session.
+
+```jsx
+<Canvas vr />
+
+// Can now be
+
+<Canvas />
+```
+
+## Automatic concurrency
+
+The `mode` prop was removed with React 18, which automatically switches between blocking (default) and concurrent (async).
+
+```jsx
+<Canvas mode="blocking" />
+
+// Can now be
+
+<Canvas />
+```
+
+You can enable concurrency with the new `startTransition` and `useTransition` APIs.
+
+```jsx
+import { useTransition } from 'react'
+import { Points } from '@react-three/drei'
+
+const [startTransition, isPending] = useTransition()
+const [radius, setRadius] = useState(1)
+const positions = calculatePositions(radius)
+const colors = calculateColors(radius)
+const sizes = calculateSizes(radius)
+
+<Points
+  positions={positions}
+  colors={colors}
+  sizes={sizes}
+  onPointerOut={() => {
+    startTransition(() => {
+      setRadius(prev => prev + 1)
+    })
+  }}
+>
+  <meshBasicMaterial vertexColors />
+</Points>
+```
+
+## Conditional rendering with frameloop
+
+`frameloop` can now be toggled to render conditionally. This is useful to toggle on user interaction or while in frame.
+
+```jsx
+const [frameloop, setFrameloop] = useState('never')
+
+<Canvas
+  frameloop={frameloop}
+  onClick={() => setFrameloop('always')}
+/>
+
+//
+
+const canvasRef = useRef()
+const [frameloop, setFrameloop] = useState('never')
+
+useEffect(() => {
+  const observer = new IntersectionObserver(([{ isIntersecting }]) => {
+    setFrameloop(isIntersecting ? 'always' : 'never')
+  }, {})
+
+  observer.observe(canvasRef.current)
+
+  return () => observer.disconnect()
+}, [])
+
+<Canvas ref={canvasRef} frameloop={frameloop} />
+```
+
+## Expanded gl prop
+
+The `gl` prop can now accept both constructor args and renderer properties like the `camera` prop.
+
+```jsx
+<Canvas gl={{ alpha: false, physicallyCorrectLights: true }} />
+```
+
+It can also accept a synchronous callback to manually create a renderer.
+
+```jsx
+<Canvas gl={(canvas) => new Renderer({ canvas })} />
+```
+
+## Manual camera manipulation
+
+Cameras can be controlled manually by setting manual to true in `camera`. This will opt out of projection matrix recalculation when the drawing area resizes.
+
+```jsx
+import { PerspectiveCamera } from '@react-three/drei'
+
+;<Canvas camera={{ manual: true }}>
+  <PerspectiveCamera makeDefault args={[45, 1]} />
+</Canvas>
+```
+
+## Spread Canvas props
+
+The `<Canvas />` can now accept non-render props to spread as native props.
+
+```jsx
+<div aria-describedby={...}>
+  <Canvas />
+</div>
+
+// can now be
+
+<Canvas aria-describedby={...} />
+```
+
+## New createRoot API
+
+`render` is depreciated in v8 for the new `createRoot` signature.
+
+```jsx
+import { createRoot } from '@react-three/fiber'
+
+let root
+
+window.addEventListener('resize', () =>
+  root = createRoot(document.querySelector('canvas'), {
+    events,
+    size: { width: window.innerWidth, height: window.innerHeight },
+  ).render(<mesh />)
+)
```

Comment:
```suggestion
import { createRoot, events } from '@react-three/fiber'

let root

window.addEventListener('resize', () =>
  (root = createRoot(document.querySelector('canvas'), {
    events,
    size: { width: window.innerWidth, height: window.innerHeight },
  }).render(<mesh />))
)
```

Reply:
Oh, that's a neat feature. I can just commit it from here.